tools/dds/dds-sniffer/rs-dds-sniffer.cpp

```diff
@@ -34,9 +40,11 @@ int main( int argc, char ** argv ) try
     CmdLine cmd( "librealsense rs-dds-sniffer tool", ' ' );
     SwitchArg snapshot_arg( "s", "snapshot", "run momentarily taking a snapshot of the domain" );
     SwitchArg machine_readable_arg( "m", "machine-readable", "output entities in a way more suitable for automatic parsing" );
+    SwitchArg topic_samples_arg( "t", "topic-samples", "register to topics that send TypeObject and print their samples" );
```

Comment:
How will it look like for the user?, if we have 15 topics with lots of fields we print all fields of all topics?
Maybe we should think of a way that the user can choose the topic he want to echo?

We can even in the non `snapshot` mode let the user interact like `ROS2`.
command like
`topic list`
`topic echo /realsense/device-info`

Not for now but worth thinking about :)